Classes/Middleware/ElementBodyResponseMiddleware.php

```diff
@@ -106,6 +106,30 @@ private function extractElement(array $content, int $elementId, bool $recursiveE
                 }
             }
         }
+        
+        // If no content found then its possible that option doNotGroupByColPos = 1 is set
+        // then try to find without groups of colpos
+        foreach ($content as $item) {
+            if (!is_array($item)) {
+                continue;
+            }
+
+            if ((int)($item['id'] ?? 0) === $elementId) {
```

Comment:
I think we can add this condition in first iteration in content to check if we have flat array

Reply:
and then we can skip whole additional iterations